Gems/Atom/Feature/Common/Code/Source/CoreLights/CapsuleLightFeatureProcessor.h

```diff
@@ -55,10 +56,13 @@ namespace AZ
         private:
             CapsuleLightFeatureProcessor(const CapsuleLightFeatureProcessor&) = delete;
 
+            void UpdateBounds(LightHandle handle);
+
             static constexpr const char* FeatureProcessorName = "CapsuleLightFeatureProcessor";
 
-            IndexedDataVector<CapsuleLightData> m_capsuleLightData;
+            MultiIndexedDataVector<CapsuleLightData, AZ::Sphere> m_lightData;
```

Comment:
So MultiIndexedDataVector stores multiple separate vectors per type. The CapsuleLightData is the data sent to the shader, the AZ::Sphere is just used for bounds checks on the CPU. If I were to mix them, I'd have to send a bunch of data to the GPU that doesn't belong there. 

I'll admit some of the Getter functions could be nicer. I'd rather have `GetData<Type>()` instead of using an index. That may be possible now (I think when this class was written we were on an older version of c++).

Reply:
I think this is fixable at the MultiIndexedDataVector level, to allow GetData by type instead of index, but I need to look into it some more. I'd prefer to push that piece of work out to a separate PR.